modules/sdk-coin-sol/src/sol.ts

```diff
@@ -557,58 +557,80 @@ export class Sol extends BaseCoin {
    */
   async recover(params: RecoveryOptions): Promise<SolTx> {
     let isDurableNonceRecovery = false;
+    let isUnsignedSweep = false;
+    let userSigningMaterial;
+    let backupSigningMaterial;
+    let userKey;
+    let backupKey;
+
+    if (!params.userKey && !params.backupKey) {
+      isUnsignedSweep = true;
+      isDurableNonceRecovery = true;
 
-    if (!params.userKey) {
-      throw new Error('missing userKey');
+      if (!params.durableNoncePK || !params.durableNonceSK) {
+        throw new Error('missing nonce account for unsigned sweep');
+      }
     }
 
-    if (!params.backupKey) {
-      throw new Error('missing backupKey');
-    }
+    if (!isUnsignedSweep) {
```

Comment:
You have this `isUnsignedSweep` check in multiple places, which makes it pretty difficult to follow the code.
you can shrink this down to a single check if you move it at the very end

An outline of your code should look something like:

1. derive Solana address from `params.bitgoKey`
2. build unsigned transaction; this is where you need to calculate transfer amount and figure out if you need to get a recent blockhash or one from durable nonce
3. finally, if this isn't for unsigned sweep, generate TSS signature with user & backup key and attach to transaction
4. return transaction in broadcast format